notebook/notebookapp.py

```diff
@@ -300,7 +306,7 @@ def init_settings(self, jupyter_app, kernel_manager, contents_manager,
             allow_password_change=jupyter_app.allow_password_change,
             server_root_dir=root_dir,
             jinja2_env=env,
-            terminals_available=False,  # Set later if terminals are available
+            terminals_available=terminado_available and terminals_enabled,
```

Comment:
It might be neater to use `jupyter_app.terminals_enabled` to save having to pass the extra parameter through.